pkg/component/openvpnclient.go

```diff
@@ -1,97 +1,242 @@
 package component
 
 import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
 	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
 	"log"
 	"net"
 	"os"
 	"path/filepath"
+	"text/template"
+	"time"
 
 	certutil "k8s.io/client-go/util/cert"
-	pki "k8s.io/kubernetes/cmd/kubeadm/app/util/pkiutil"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/pkiutil"
 )
 
 const (
-	OpenVPNClientCert   = "client.crt"
-	OpenVPNClientKey    = "client.key"
-	OpenVPNClientCaCert = "ca.crt"
-	OpenVPNClientCaKey  = "ca.key"
-	OpenVPNClinetTaKey  = "ta.key"
+	OpenVPNClientCert        = "client.crt"
+	OpenVPNClientKey         = "client.key"
+	OpenVPNClientCaCert      = "ca.crt"
+	OpenVPNClientCaKey       = "ca.key"
+	OpenVPNClientTaKey       = "ta.key"
+	OpenVPNClientPath        = "pki/vpn-client"
+	OpenVPNClientRevokedPath = "pki/vpn-client/revoked"
+
+	openVPNClientConfigTmpl = `
+client
+dev tun
+proto udp
+remote-random
+remote-cert-tls server
+tls-version-min 1.2
+tls-cipher TLS-ECDHE-RSA-WITH-AES-128-GCM-SHA256:TLS-ECDHE-ECDSA-WITH-AES-128-GCM-SHA256:TLS-ECDHE-RSA-WITH-AES-256-GCM-SHA384:TLS-DHE-RSA-WITH-AES-256-CBC-SHA256
+cipher AES-256-CBC
+auth SHA512
+resolv-retry infinite
+auth-retry none
+nobind
+key-direction 1
+persist-key
+persist-tun
+compress lzo
+verb 3
+tls-client
+
+{{ range $server := .Server }}remote {{ $server }} 1194
+{{ end }}
+
+<ca>
+{{ .CACert }}</ca>
+
+<cert>
+{{ .ClientCert }}</cert>
+
+<key>
+{{ .ClientKey }}</key>
+
+<tls-auth>
+{{ .TLSAuthKey }}</tls-auth>
+`
 )
```

Comment:
This would fix https://github.com/sighupio/fury-kubernetes-openvpn/issues/2 🎉

Reply:
Yes, I've updated the commit message to close the issue once the PR gets merged.